typescript/test/util.ts

```diff
@@ -0,0 +1,115 @@
+import * as assert from 'assert';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+const NO_OUTPUT_FILE = 0;
+const IDENTICAL_FILE = 1;
+const FILE_WITH_DIFF_CONTENT = 2;
+
+const BASELINE_EXTENSION = '.baseline';
+export function compareToBaseline(outpurDir: string, baselineDir: string) {
+  // put all baseline files into fileStack
+  let fileStack: string[] = [];
+  const dirStack: string[] = [];
+  putAllBaselineFiles(baselineDir, fileStack, dirStack);
+  // store every item (file or directory with full path in output dir and baseline dir) in stack, pop once a time
+  const protoItemStack: Item[] = [];
+  putItemToStack(protoItemStack, outpurDir, baselineDir);
+  while (protoItemStack.length !== 0) {
+    const item = protoItemStack[protoItemStack.length - 1];
+    protoItemStack.pop();
+    // if item is a file, compare it with baseline
+    if (fs.lstatSync(item.outputPath).isFile()) {
+      const compareResult = checkIdenticalFile(
+        item.outputPath,
+        item.baselinePath + BASELINE_EXTENSION
+      );
+      // if two files are identical or it's generated properly, filter it from the stack.
+      if (
+        compareResult === IDENTICAL_FILE ||
+        compareResult === NO_OUTPUT_FILE
+      ) {
+        fileStack = fileStack.filter(
+          file => file !== item.baselinePath + BASELINE_EXTENSION
+        );
+      }
+    } else if (fs.lstatSync(item.outputPath).isDirectory()) {
+      // if item is a directory, loop the folder and put every item in stack again
+      putItemToStack(protoItemStack, item.outputPath, item.baselinePath);
+    }
+  }
+  if (fileStack.length !== 0) {
+    fileStack.forEach(file => {
+      console.error(file + ' is not identical with the generated file. ');
+    });
+  }
+  assert.strictEqual(fileStack.length, 0);
+}
+
+function checkIdenticalFile(
+  outputFullPath: string,
+  baselineFullPath: string
+): number {
+  if (!outputFullPath.endsWith('.proto')) {
+    assert(fs.existsSync(baselineFullPath));
```

Comment:
Let's not assert this, because it stops other files from being checked. Let's just return `IDENTICAL_FILE` to skip the check for `.proto` files.

Reply:
Sounds good, take proto file as identical and log error if the file does bot exist. Return boolean for the util method `equalToBaseline` and only assert there. Thanks!